### ISSUE TITLE
Ensure root a11y view is not redundantly invalidated on Android

### ISSUE THREAD

**Issue by nscobie** [flutter/flutter#63823]
Requested tags: accessibility, android, engine, (and maybe others).

According to [this code in AccessibilityBridge.java](https://github.com/flutter/engine/blob/dbf7e831199d1af7f9f5e334a08b9cca669fd119/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L943), it looks like the `rootAccessibilityView` should be manually [invalidated ](https://developer.android.com/reference/android/view/View#invalidate()) when a node is focused, **if** there was no previously focused node.

According to the comment at the code linked above:

> When Android focuses a node, it doesn't invalidate the view.
> (It does when it sends ACTION_CLEAR_ACCESSIBILITY_FOCUS, so
> we only have to worry about this when the focused node is null.)

However, before this code is reached, the `TYPE_VIEW_ACCESSIBILITY_FOCUSED` [event is sent](https://github.com/flutter/engine/blob/dbf7e831199d1af7f9f5e334a08b9cca669fd119/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L941), which will trigger `ACTION_CLEAR_ACCESSIBILITY_FOCUS`, which both sets `accessibilityFocusedSemanticsNode` [to null](https://github.com/flutter/engine/blob/dbf7e831199d1af7f9f5e334a08b9cca669fd119/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L933) and invalidates the `rootAccessibilityView` (according to the comment above) before this code runs. Since `accessibilityFocusedSemanticsNode` will be set to null, this condition should always pass, which sounds like it will make a redundant call to `rootAccessibilityView.invalidate()`.

I am unsure of performance impact, and whether or not there are situations where sending `TYPE_VIEW_ACCESSIBILITY_FOCUSED` might not trigger Android/TalkBack to respond by sending `ACTION_CLEAR_ACCESSIBILITY_FOCUS`, or if there are situations where `ACTION_CLEAR_ACCESSIBILITY_FOCUS` does not invalidate the root view. Further investigation is required.

I no longer have access to the testing conditions where I discovered this anomaly, so I can't include output from `flutter doctor -v`.

Device: `Pixel 3a`
Flutter version: `approximately 1.21`

**Comment by TahaTesser** [comment]
Hi @nscobie 
Can you please create testing condition `Pixel 3a` if you have ? and  provide a complete reproducible minimal code sample
Thank you
Thank you

**Comment by nscobie** [comment]
Hey @TahaTesser, apologies for the delay.

To clarify, this isn't necessarily a concrete issue, but more of an anomaly I noticed where redundant work might be occurring, but needs more investigation.

Unfortunately I no longer have access to the testing conditions to run `flutter doctor`. The sample app I was using was from #36307, although I don't believe the app used to reproduce should matter.

**Comment by TahaTesser** [comment]
I’m unable to reproduce the issue to pinpoint the cause but this could be worth looking into. The issue needs further investigation.
